src/main/java/pl/damianszczepanik/jenkins/buildhistorymanager/model/actions/DeleteArtifactsWithPatternAction.java

```diff
@@ -0,0 +1,175 @@
+        } else {
+            return archiveRootPath;
+        }
+    }
+
+    public static boolean isDirEmpty(Path path) throws IOException {
+        if (Files.isDirectory(path)) {
+            try (Stream<Path> entries = Files.list(path)) {
+                return !entries.findFirst().isPresent();
+            }
+        }        
+        return false;
+    }
+
+    // if 'file' is on a different node, this FileCallable will be transferred to that node and executed there.
+    public static final class Delete implements FileCallable<Void> {
+        private static final long serialVersionUID = 1;
+        private final String archiveRootPath;
+
+        public Delete(String archiveRootPath) {
+            this.archiveRootPath = removeLastSlash(archiveRootPath);
+        }
+
+        @Override 
+        public Void invoke(File vFile, VirtualChannel channel) throws IOException {          
+
+            Set<File> directories = new HashSet<>();
+            deleteFileOrLogError(vFile, directories);
+            deleteEmptyDirectoriesAndParents(directories);
+            return null;
+        }
+
+        public void deleteEmptyDirectoriesAndParents(Set<File> directories) throws IOException {
+            for (File dir : directories){
+                if (shouldDeleteDirectory(dir)) {
+                    Util.deleteFile(dir);
+                    deleteParentDirectories(dir.getParentFile());
+                }
+            }
+        }
+
+        public boolean shouldDeleteDirectory(File dir) throws IOException {
+            if (isInvalidDirectory(dir)) {
+                return false;
+            }
+
+            File parent = dir.getParentFile();
+            return hasValidParent(parent);
+        }
+
+        public boolean hasValidParent(File parent) {
+            return parent != null && parent.getPath().equals(this.archiveRootPath);
+        }
+
+        public boolean isInvalidDirectory(File dir) throws IOException {
+            return dir == null || !dir.isDirectory() || !isDirEmpty(dir.toPath());
+        }
+
+        public void deleteParentDirectories(File directory) {
+            File parent = directory;
+
+            while (shouldDelete(parent)) {
+                deleteDirectory(parent);
+                parent = parent.getParentFile();
+            }
+        }
+
+        public boolean shouldDelete(File directory) {
+            return directory != null && !directory.getPath().equals(this.archiveRootPath);
+        }
+
+        public void deleteDirectory(File directory) {
+            boolean deleteSuccess = directory.delete();
+            if(!deleteSuccess) {
+                throw new RuntimeException("Deletion of directory failed: " + directory.getAbsolutePath());
+            }
+        }
+
+        public void deleteFileOrLogError(File vFile, Set<File> directories) throws IOException {
+            if (vFile.isFile()) {
+                LOGGER.log(Level.FINE, "Deleting " + vFile.getName());
```

Comment:
LOGGER -> LOG
so this meet convention which you can find in other classes

Reply:
Renamed `LOGGER` to 'LOG' for naming convension.